### PR TITLE
K8SPG-371: Use a different API group for v2

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPG-371](https://jira.percona.com/browse/K8SPG-371) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
*Short explanation of the problem.*

**Cause:**
*Short explanation of the root cause of the issue if applicable.*

**Solution:**
*Short explanation of the solution we are providing with this PR.*

**CHECKLIST**
---
**Jira**
- [ ] Is the Jira ticket created and referenced properly?
- [ ] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [ ] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PG version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/e620400bf/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/e620400bf/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/e620400bf/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/e620400bf/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/e620400bf/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/e620400bf/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/e620400bf/telemetry-transfer.log) |
| We run 7 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/450/commits/e620400bf097d58b4c9c2bfce267be44c4ced8fc
image: `perconalab/percona-postgresql-operator:PR-450-e620400bf`

File: config/bundle/kustomization.yaml [hunk 1]
```diff
@@ -7,4 +7,4 @@ resources:
 images:
 - name: postgres-operator
   newName: perconalab/percona-postgresql-operator
-  newTag: main
+  newTag: k8spg-355
```
Comment (hors) [inline config/bundle/kustomization.yaml:10]: please fix it

File: config/manager/namespace/kustomization.yaml [hunk 2]
```diff
@@ -10,4 +10,4 @@ patchesStrategicMerge:
 images:
 - name: postgres-operator
   newName: perconalab/percona-postgresql-operator
-  newTag: main
+  newTag: k8spg-355
```
Comment (hors) [inline config/manager/namespace/kustomization.yaml:13]: please fix

File: deploy/cw-bundle.yaml [hunk 3]
```diff
@@ -9,9 +9,9 @@ metadata:
   annotations:
```
Comment (hors) [inline deploy/cw-bundle.yaml:9]: @egegunes can we remove the namespace feald for CW under this PR as we discussed yesterday

**Review by hors — CHANGES_REQUESTED** [review]

File: deploy/cw-bundle.yaml (hunk 3 above)
Comment (egegunes) [reply deploy/cw-bundle.yaml:9]: https://github.com/percona/percona-postgresql-operator/pull/450/commits/dea752c001f0eaa1975407db476d038be4b2ef95

File: config/manager/namespace/kustomization.yaml (hunk 2 above)
Comment (egegunes) [reply config/manager/namespace/kustomization.yaml]: https://github.com/percona/percona-postgresql-operator/pull/450/commits/dea752c001f0eaa1975407db476d038be4b2ef95

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/5724755ef/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/5724755ef/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/5724755ef/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/5724755ef/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/5724755ef/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/5724755ef/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/5724755ef/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/5724755ef/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/450/commits/5724755eff87bd203ce6826ed2ec608cc8586f27
image: `perconalab/percona-postgresql-operator:PR-450-5724755ef`

**Review by hors — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/450/commits/2ffe63c305645df9a74dd88c657b05605c6aa62a
image: `perconalab/percona-postgresql-operator:PR-450-2ffe63c30`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/2ffe63c30/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/2ffe63c30/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/2ffe63c30/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/2ffe63c30/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/2ffe63c30/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/2ffe63c30/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/2ffe63c30/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/2ffe63c30/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/450/commits/2ffe63c305645df9a74dd88c657b05605c6aa62a
image: `perconalab/percona-postgresql-operator:PR-450-2ffe63c30`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/450/commits/caef82c108675afae2b4f0dc4f76cbfcbe92c6e9
image: `perconalab/percona-postgresql-operator:PR-450-caef82c10`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-450/caef82c10/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/450/commits/caef82c108675afae2b4f0dc4f76cbfcbe92c6e9
image: `perconalab/percona-postgresql-operator:PR-450-caef82c10`

**Review by hors — APPROVED** [review]

**Pull request merged**